### PR TITLE
Drop 2.319.x

### PR DESCRIPTION
**Pull request by timja**
This is becoming a large maintenance burden.

So many pinned plugins on this line and lots are failing due to it.

Let's drop it.

See also https://github.com/jenkinsci/bom/pull/1063#issuecomment-1297174680

### CODE REVIEW

**Review by jetersen — APPROVED** [review]

**Comment by jglick** [comment]
Hmm, looks like 2.332.x is not in great shape!

**Comment by timja** [comment]
> Started
java.lang.NoSuchMethodError: 'boolean org.kohsuke.groovy.sandbox.SandboxTransformer.mightBePositionalArgumentConstructor(org.codehaus.groovy.ast.expr.VariableExpression)'
 at com.cloudbees.groovy.cps.SandboxCpsTransformer.visitAssignmentOrCast(SandboxCpsTransformer.java:93)
 at com.cloudbees.groovy.cps.CpsTransformer$42.run(CpsTransformer.java:1171)
 at com.cloudbees.groovy.cps.CpsTransformer.makeChildren(CpsTransformer.java:446)
 at com.cloudbees.groovy.cps.CpsTransformer.makeNode(CpsTransformer.java:409)
 at com.cloudbees.groovy.cps.CpsTransformer.visitDeclarationExpression(CpsTransformer.java:1164)
 at org.codehaus.groovy.ast.expr.DeclarationExpression.visit(DeclarationExpression.java:89)
 at com.cloudbees.groovy.cps.CpsTransformer.visit(CpsTransformer.java:357)
 at com.cloudbees.groovy.cps.CpsTransformer.visitExpressionStatement(CpsTransformer.java:589)
 at org.codehaus.groovy.ast.stmt.ExpressionStatement.visit(ExpressionStatement.java:42)

Mean anything to anyone? cc @dwnusbaum ?

**Comment by dwnusbaum** [comment]
Yeah probably, I will take a look. Why would it not have failed in #1507 though?

**Comment by dwnusbaum** [comment]
Actually no, `workflow-cps` was already frozen on the 332.x line: https://github.com/dwnusbaum/bom/blob/f2ca765c169a9ee18ba92b4096d19eab2f2ee169/bom-2.332.x/pom.xml#L136, that's why I didn't freeze it in my PR.

**Comment by dwnusbaum** [comment]
I think https://github.com/jenkinsci/bom/pull/1512 is the problem (Dependabot specifically filed a PR to update the 332.x line). I do not understand how that passed CI though.

**Comment by jglick** [comment]
> some way to configure updatecli to have it ignore everything besides `bom-weekly/`

So updatecli is being used for two things: core and plugins.

In the case of core, it updates the weekly version, which Dependabot cannot do. That gets tested, so fine.

In the case of plugins, Dependabot handles `bom-weekly` but the whole point of using updatecli is to propose updates to plugins otherwise pinned on LTS lines because of version constraints, which Dependabot cannot handle. Yet to avoid swamping ci.jenkins.io we only test weekly plus the oldest line, so the proposed updates on intermediate lines are not guaranteed to be correct at all.

We could run updatecli only on the oldest line, though it is actually the newer lines where it is more useful.

Not sure what to do besides just removing the updatecli plugin code. It does not work as nicely as Dependabot anyway. We do not really need this functionality.

**Comment by jglick** [comment]
<details>
<summary>Some sort of jakarta-mail problem</summary>

```
FATAL: jakarta.mail.Provider: com.sun.mail.imap.IMAPProvider not a subtype
java.util.ServiceConfigurationError: jakarta.mail.Provider: com.sun.mail.imap.IMAPProvider not a subtype
	at java.base/java.util.ServiceLoader.fail(ServiceLoader.java:589)
	at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNextService(ServiceLoader.java:1237)
	at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNext(ServiceLoader.java:1265)
	at java.base/java.util.ServiceLoader$2.hasNext(ServiceLoader.java:1300)
	at java.base/java.util.ServiceLoader$3.hasNext(ServiceLoader.java:1385)
	at jakarta.mail.Session.loadProviders(Session.java:964)
	at jakarta.mail.Session.<init>(Session.java:254)
	at jakarta.mail.Session.getInstance(Session.java:281)
	at hudson.tasks.Mailer$DescriptorImpl.createSession(Mailer.java:415)
	at hudson.tasks.Mailer$DescriptorImpl.createSession(Mailer.java:353)
	at jenkins.plugins.mailer.tasks.MimeMessageBuilder.buildMimeMessage(MimeMessageBuilder.java:199)
	at hudson.tasks.MailSender.createEmptyMail(MailSender.java:407)
	at hudson.tasks.MailSender.createFailureMail(MailSender.java:274)
	at hudson.tasks.MailSender.createMail(MailSender.java:182)
	at hudson.tasks.MailSender.run(MailSender.java:112)
	at hudson.tasks.Mailer.perform(Mailer.java:175)
	at hudson.tasks.Mailer.perform(Mailer.java:138)
```

</details>

**Comment by jglick** [comment]
And it is not until https://github.com/jenkinsci/mailer-plugin/pull/183 (as yet unreleased) that `dependency:tree` omits `javax.mail`. Why would it matter? None of `workflow-basic-steps`, `javax-mail-api`, and `jakarta-mail-api` are pinned, and https://github.com/jenkinsci/mailer-plugin/compare/435.v79ef3972b_5c7...438.v02c7f0a_12fa_4 only updates the BOM import which should not be relevant since PCT testing overrides everything from the current megawar.

**Comment by jglick** [comment]
`mailer` also broken in 2.346.x but working in 2.361.x so I suppose the version bump in #1291 mattered somehow.

File: failFast [hunk 1]
```diff
@@ -1 +1 @@
-true
+false
```
Comment (jglick) [inline failFast:1]: (temporarily)

File: bom-2.346.x/pom.xml [hunk 2]
```diff
@@ -40,7 +40,7 @@
             <dependency>
                 <groupId>org.jenkins-ci.plugins</groupId>
                 <artifactId>mailer</artifactId>
-                <version>435.v79ef3972b_5c7</version>
+                <version>435.437.vec1ef333b_df7</version> <!-- TODO https://github.com/jenkinsci/mailer-plugin/pull/201 -->
```
Comment (jglick) [inline bom-2.346.x/pom.xml:43]: https://github.com/jenkinsci/mailer-plugin/pull/201

**Comment by jglick** [comment]
```
 [ERROR] Failed to execute goal org.jenkins-ci.tools:maven-hpi-plugin:3.33:resolve-test-dependencies (default-resolve-test-dependencies) on project antisamy-markup-formatter: Unable to resolve dependencies: Could not resolve dependencies for project org.jenkins-ci.plugins:antisamy-markup-formatter:hpi:2.7: Could not find artifact org.jenkins-ci.plugins:mailer:jar:435.437.vec1ef333b_df7 in repo.jenkins-ci.org (https://repo.jenkins-ci.org/public/) -> [Help 1]
```

That plugin is JEP-305 (in fact JEP-229) enabled, so I am not sure why it is not looking in `incrementals`. (PCT passed locally with my personal Maven settings.) :shrug: will assume that is somehow related to using a prerelease version.

**Pull request merged**